### ISSUE TITLE
An Entity Hidden With Camera.Main.ExcludeTags Is Hidden Even After The Tag Is Removed From The Entity

### ISSUE THREAD

**Issue by anthonysharpy** [sboxgame/issues#3490]
### Describe the bug

i am setting a tag on the player, their clothes and their weapons that represents their steamid. i'll just show the code for the weapon side of things here:

```
// Taking the weapon out of our pocket.
public virtual void OnEquip()
{
	Game.AssertServer();

	Tags.Add( Client.GetSteamIDAsString() );
}
```

```
// We can't have numbers in entity tags so we are converting it to letters.
public static string GetSteamIDAsString( this IClient client )
{
	if ( client == null )
		return null;

	var id = client.SteamId.ToString();

	id = id.Replace( '1', 'a' )
		.Replace( '2', 'b' )
		.Replace( '3', 'c' )
		.Replace( '4', 'd' )
		.Replace( '5', 'e' )
		.Replace( '6', 'f' )
		.Replace( '7', 'g' )
		.Replace( '8', 'h' )
		.Replace( '9', 'i' )
		.Replace( '0', 'j' );

	return id;
}
```

if you're wondering "umm... why?" it's because it's my workaround/solution for this camera rendering issue: https://github.com/sboxgame/issues/issues/3147

this all gets hidden/unhidden depending on whether the player is in first or third person, like this (ignore the crappy code - i rushed lol 😄 ):

```
public void DoCamera()
{
	if ( CameraInThirdPersonMode )
	{
		if ( Camera.Main.ExcludeTags.Has( Game.LocalClient.GetSteamIDAsString() ) )
			Camera.Main.ExcludeTags.Remove( Game.LocalClient.GetSteamIDAsString() );

		...
	}
	else
	{
		if ( !Camera.Main.ExcludeTags.Has( Game.LocalClient.GetSteamIDAsString() ) )
			Camera.Main.ExcludeTags.Add( Game.LocalClient.GetSteamIDAsString() );

		...
	}
}
```

when the player drops their weapon, this steam id tag is removed from the weapon:

```
// Dropping the weapon from our inventory. Called before it's actually dropped.
public void OnDrop()
{
	Game.AssertServer();

	Tags.Remove( Client.GetSteamIDAsString() );
}
```

the issue is that even after this tag is removed from the weapon, the weapon is still being ignored by the camera. it's as if the tag wasn't actually removed. 

here's a video. as you can see, before dropping it the pistol has the tag, but after dropping it doesn't. after being dropped, it should be visible in first and third person but for some reason it's only visible in third person.

https://github.com/sboxgame/issues/assets/55365296/a9df57e9-33f4-4aa5-8816-aa5facd300d1

hopefully this is not just me getting confused.

TLDR: if you hide a tag with Camera.Main.ExcludeTags, and then you remove the tag from an entity, the entity is still hidden

### To Reproduce

see above

### Expected behavior

if you hide a tag with Camera.Main.ExcludeTags, and then you remove the tag from the entity, the entity should be visible again
